File: app/src/main/java/br/eng/ecarrara/vilibra/domain/repository/BookRepository.java
package br.eng.ecarrara.vilibra.domain.repository;

import java.util.List;

import br.eng.ecarrara.vilibra.domain.entity.Book;

/**
 * Defines an interface to provide access to a repository of {@link Book}
 */
public interface BookRepository {

    Book byIsbn(final String isbn);
    void add(final Book book);

}

File: app/src/main/kotlin/br/eng/ecarrara/vilibra/book/domain/BookRepository.kt
package br.eng.ecarrara.vilibra.book.domain

import br.eng.ecarrara.vilibra.book.domain.entity.Book
import io.reactivex.Completable
import io.reactivex.Single

/**
 * Defines an interface to provide access to a repository of [Book]
 */
interface BookRepository {

    fun getByIsbn(isbn: String): Single<Book>
    fun add(book: Book): Completable

}
